### PR TITLE
type: optimization undefined type

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | type: optimization undefined type |
| 🇨🇳 Chinese | 优化 undefined 类型 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by li-jia-nan** [comment]
按照 ts 的规范，把可选参数放在最后了

File: components/anchor/demo/targetOffset.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Anchor, Row, Col } from 'antd';
 
 const App: React.FC = () => {
   const topRef = React.useRef<HTMLDivElement>(null);
-  const [targetOffset, setTargetOffset] = useState<number | undefined>(undefined);
+  const [targetOffset, setTargetOffset] = useState<number>();
```
Comment (zombieJ) [inline components/anchor/demo/targetOffset.tsx:6]: 原来这个也可以默认 `undefined` 了（知识库已更新~）

File: components/badge/SingleNumber.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ export interface UnitNumberProps {
 }
 
 function UnitNumber({ prefixCls, value, current, offset = 0 }: UnitNumberProps) {
-  let style: React.CSSProperties | undefined;
+  let style: React.CSSProperties = {};
```
Comment (zombieJ) [inline components/badge/SingleNumber.tsx:12]: 这个不加 `{}`，运行时多了申明对象的损耗。下同

File: components/anchor/demo/targetOffset.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/anchor/demo/targetOffset.tsx:6]: image

File: components/date-picker/generatePicker/generateRangePicker.tsx [hunk 3]
```diff
@@ -117,7 +117,7 @@ export default function generateRangePicker<DateType>(generateConfig: GenerateCo
               disabled={mergedDisabled}
               ref={innerRef}
               dropdownAlign={transPlacement2DropdownAlign(direction, placement)}
-              placeholder={getRangePlaceholder(picker, locale, placeholder)}
+              placeholder={getRangePlaceholder(locale, picker, placeholder)}
```
Comment (zombieJ) [inline components/date-picker/generatePicker/generateRangePicker.tsx:120]: 避免改 param 顺序，用户可能直接用了内部方法。虽然我们不用保证这个，但是如果是改 ts 定义则尽量不做 break。

File: components/badge/SingleNumber.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/badge/SingleNumber.tsx]: revert~

File: components/date-picker/generatePicker/generateRangePicker.tsx (hunk 3 above)
Comment (li-jia-nan) [reply components/date-picker/generatePicker/generateRangePicker.tsx]: revert

**Review by zombieJ — APPROVED** [review]

**Pull request merged**